crates/sui-graphql-rpc/src/types/object_change.rs

```diff
@@ -2,87 +2,54 @@
 // SPDX-License-Identifier: Apache-2.0
 
 use async_graphql::*;
-use sui_indexer::types_v2::IndexedObjectChange;
+use sui_types::effects::ObjectChange as NativeObjectChange;
 
 use super::{object::Object, sui_address::SuiAddress};
-use crate::{context_data::db_data_provider::PgManager, error::Error};
+use crate::context_data::db_data_provider::PgManager;
 
 pub(crate) struct ObjectChange {
-    // TODO: input_key (waiting for object history)
-    output_key: Option<(SuiAddress, u64)>,
-    id_created: bool,
-    id_deleted: bool,
+    pub native: NativeObjectChange,
 }
 
 /// Effect on an individual Object (keyed by its ID).
 #[Object]
 impl ObjectChange {
-    /// The contents of the object at the end of the transaction.
+    /// The address of the object that has changed.
+    async fn location(&self) -> SuiAddress {
+        self.native.id.into()
+    }
+
+    /// The contents of the object immediately before the transaction.
+    async fn input_state(&self, ctx: &Context<'_>) -> Result<Option<Object>> {
```

Comment:
This implementation will still always return `null` until we have object history, but once we do, it will be correct in all cases.